ValveResourceFormat/Resource/ResourceTypes/KV3.cs

```diff
@@ -0,0 +1,592 @@
+        {
+            //Initialise datastructures
+            objStack = new Stack<KVObject>();
+            stateStack = new Stack<State>();
+            stateStack.Push(State.HEADER);
+
+            root = new KVObject("root");
+            objStack.Push(root);
+
+            currentString = new StringBuilder();
+
+            //Open stream
+            fileStream = File.ReadAllText(filename).ToCharArray();
+            index = 0;
+
+            char c;
+            while (index < fileStream.Length)
+            {
+                //Read character
+                c = fileStream[index];
+
+                //Do something depending on the current state
+                switch (stateStack.Peek())
+                {
+                    case State.HEADER:
+                        ReadHeader(c);
+                        break;
+                    case State.PROP_NAME:
+                        ReadPropName(c);
+                        break;
+                    case State.SEEK_VALUE:
+                        SeekValue(c);
+                        break;
+                    case State.VALUE_STRUCT:
+                        ReadValueStruct(c);
+                        break;
+                    case State.VALUE_STRING:
+                        ReadValueString(c);
+                        break;
+                    case State.VALUE_STRING_MULTI:
+                        ReadValueStringMulti(c);
+                        break;
+                    case State.VALUE_BOOL:
+                        ReadValueBool(c);
+                        break;
+                    case State.VALUE_NUMBER:
+                        ReadValueNumber(c);
+                        break;
+                    case State.VALUE_ARRAY:
+                        ReadValueArray(c);
+                        break;
+                    case State.VALUE_FLAGGED:
+                        ReadValueFlagged(c);
+                        break;
+                    case State.COMMENT:
+                        ReadComment(c);
+                        break;
+                    case State.COMMENT_BLOCK:
+                        ReadCommentBlock(c);
+                        break;
+                }
+
+                //Advance read index
+                index++;
+            }
+
+            return new KVFile(root.Properties.ElementAt(0).Key, (KVObject)root.Properties.ElementAt(0).Value.Value);
+        }
+
+        //header state
+        private static void ReadHeader(char c)
+        {
+            currentString.Append(c);
+
+            //Read until --> is encountered
+            if (c == '>' && currentString.ToString().Substring( currentString.Length - 3 ) == "-->") {
+                stateStack.Pop();
+                stateStack.Push(State.SEEK_VALUE);
+                return;
+            }
+        }
+
+        //Seeking value state
+        private static void SeekValue(char c)
+        {
+            //Ignore whitespace
+            if (whiteSpace.Contains(c) || c == '=' )
```

Comment:
`char.IsWhiteSpace` already exists. Is there some reason that you need to reimplement this?